packages/block-library/src/image/image.js

```diff
@@ -61,6 +61,27 @@ function getFilename( url ) {
 	}
 }
 
+/**
+ * Checks if the given block is registered and is in the allowed blocks list.
+ *
+ * @param {string}        name Block name.
+ * @param {boolean|Array} list Allowed block types.
+ *
+ * @return {boolean}           Whether the block exists.
+ */
+function checkBlockExists( name, list ) {
```

Comment:
If this is formalized in a function, it seems like it belongs in the block/api/registration set and not on the block

Reply:
Hi, @mtias

Do you want me to create a new issue for this function?

Reply:
I think it'd make sense in order to keep the image block file cleaner. Alternatively, we might want to approach this slightly different and look at `getPossibleBlockTransformations` instead and not to the block registry.

Reply:
Yes relying on the transforms seems to be a really good idea, it may cover other checks that we may be missing. I guess we can rely on the getBlockTransformItems selector.

Reply:
Just to be clear, the `getPossibleBlockTransformations` selector already exists :)

Reply:
I agree. Using a transforms selector makes more sense here. Also, we don't have all cases covered yet - #32520.

I will create a new PR that uses `getPossibleBlockTransformations` if possible or similar logic.